### PR TITLE
SDKS-2978: Integration tests for my segments v2

### PR DESCRIPTION
**Pull request by javrudsky**
# Android SDK
- Fixed some issues parsing my segments v2 notifications types
- Added integration tests for my segments v2

### CODE REVIEW

File: src/sharedTest/java/helper/TestingData.java [hunk 1]
```diff
@@ -27,21 +46,39 @@
       8492584437244343049 11382796718859679607 11383137936375052427 17699699514337596928 17001541343685934583 8355202062888946034]
      */
     public final static String BOUNDED_NOTIFICATION_GZIP = "{" +
-            "\"type\": 1," +
+            "\"type\": \"MY_SEGMENTS_UPDATE_V2\"," +
+            "\"updateStrategy\": 1," +
```
Comment (EmilianoSanchez) [inline src/sharedTest/java/helper/TestingData.java:50]: compression should be 1 for GZIP

File: src/sharedTest/java/helper/TestingData.java [hunk 2]
```diff
@@ -27,21 +46,39 @@
...
             "\"compression\": 2," +
             "\"data\": \"H4sIAAAAAAAA/2IYBfgAx0A7YBTgB4wD7YABAAID7QC6g5EYy8MEMA20A+gMFAbaAYMZDPXqlGWgHTAKRsEoGAWjgCzQQFjJkKqiiPAPAQAIAAD//5L7VQwAEAAA\"" +
             "}";
 
+    public final static String ESCAPED_BOUNDED_NOTIFICATION_GZIP = "{" +
+            "\\\"type\\\": \\\"MY_SEGMENTS_UPDATE_V2\\\"," +
+            "\\\"updateStrategy\\\": 1," +
+            "\\\"compression\\\": 2," +
```
Comment (EmilianoSanchez) [inline src/sharedTest/java/helper/TestingData.java:58]: compression should be 1 for GZIP

**Review by EmilianoSanchez — APPROVED** [review]
LGTM, just check Javi the `compression` field in some notification mocks.

File: src/sharedTest/java/helper/TestingData.java (hunk 1 above)
Comment (javrudsky) [reply src/sharedTest/java/helper/TestingData.java:43]: Fixed!

File: src/sharedTest/java/helper/TestingData.java (hunk 2 above)
Comment (javrudsky) [reply src/sharedTest/java/helper/TestingData.java]: Fixed doc! Thank you!

**Pull request merged**